library/nameops.ml

```diff
@@ -124,34 +124,95 @@ let atompart_of_id id = fst (repr_ident id)
 
 (* Names *)
 
-let out_name = function
-  | Name id -> id
-  | Anonymous -> failwith "Nameops.out_name"
+module type ExtName =
+sig
 
-let name_fold f na a =
+  include module type of struct include Names.Name end
+
+  val fold_right : (Id.t -> 'a -> 'a) -> t -> 'a -> 'a
+  (** [fold_left f a x] is [f a id] if [x] is [Name id], and [a] otherwise. *)
+
+  val fold_left : ('a -> Id.t -> 'a) -> 'a -> t -> 'a
+  (** [fold_left f x a] is [f id a] if [x] is [Name id], and [a] otherwise. *)
+
+  val iter : (Id.t -> unit) -> t -> unit
+  (** [iter f x] does [f id] if [x] equals [Name id], nothing otherwise. *)
+
+  val map : (Id.t -> Id.t) -> t -> t
+  (** [map f x] is [Anonymous] if [x] is [Anonymous] and [Name (f id)] if [x] is [Name id]. *)
+
+  val fold_map : ('a -> Id.t -> 'a * Id.t) -> 'a -> t -> 'a * t
+  (** [fold_map f x a] is [a',Name id'] if [x] is [Name id] and [f a id] is [(a',id')]. *)
+
+  val out : t -> Id.t
+  (** [out] associates [id] to [Name id]. Raises [Failure "Nameops.out_name"] otherwise. *)
+
+  val max : t -> t -> t
+  (** [max x y] returns [x] if [x] is [Name id] and [y] otherwise. *)
+
+  val cons : t -> Id.t list -> Id.t list
+  (** [cons x l] returns [id::l] if [x] is [Name id] and [l] otherwise. *)
+
+end
+
+open Names.Name
+
+module Name =
+struct
+
+  include Names.Name
+
+  let out = function
+    | Name id -> id
+    | Anonymous -> failwith "Nameops.out_name"
```

Comment:
Certainly. How to do the transition? Change of exceptions name are not tracked by the ocaml typing system, so doing a change is (in theory) liable to introduce bugs difficult to track. So, shall we take the responsibility that documenting the change will be enough (or that noone uses the function)?